doc/ExpertOptions.md

```diff
@@ -232,13 +235,40 @@ is passed to dxCompiler using the `-input` option, it is transformed into the fo
 }
 ```
 
-The WDL specification states that a `Directory` input is to be treated as a snapshot of the directory at the time the job is executed. To enforce this behavior, at the start of the job the full (recursive) listing of the directory is retrieved, and only those files/subfolders are localized to the worker. This means that if a file is added to or removed from the directory in the DNAnexus project while the job is running, that change is not reflected in the local copy on the worker. However, if the same directory is used in multiple jobs, there is (currently) no way to guarantee that the contents are the same between workers. We strongly recommend to enact policies and practices to prevent modification of folders that will be used as input to compiled WDL workflows.
+The WDL specification states that a `Directory` input is to be treated as a snapshot of the directory at the time the 
+job is executed. To enforce this behavior, at the start of the job the full (recursive) listing of the directory is 
+retrieved, and only those files/subfolders are localized to the worker. This means that if a file is added to or removed 
+from the directory in the DNAnexus project while the job is running, that change is not reflected in the local copy on 
+the worker. However, if the same directory is used in multiple jobs, there is (currently) no way to guarantee that the 
+contents are the same between workers. We strongly recommend to enact policies and practices to prevent modification of 
+folders that will be used as input to compiled WDL workflows.
 
 A second important caveat, which results from the fact that folders are not treated as first-class objects by DNAnexus, is that, if [job reuse](#job-reuse) is enabled, a job that is run with the same folder input as a previous job (and all other inputs the same) will reuse the previous job outputs regardless of whether the contents of the folder have changed. There are two possible solutions:
 
 * Disable job reuse when running executables with `Directory`-type inputs.
 * Enact policies and practices to prevent modification of folders that will be used as input when job reuse is enabled.
 
+###### Directory outputs:
+WDL 2.0 supports directory outputs. Specify directory outputs in your tasks/workflows by providing an output pat as a string.   
+Example:
+
+```wdl
+task task_with_dir_outs {
+  input {}
+
+  command <<<
+    mkdir folderoutput
+    echo hello > folderoutput/hello.txt
+  >>>
+  output {
+    Directory outdir = 'folderoutput/'
+  }
+}
+```
+The `folderoutput/` directory has to be specified **without** `dx://` prefix or project ID. It has to exist on the worker 
```

Comment:
is project.contribute access required to run workflows and outputs with output directories?

Reply:
I did not introduce any changes to the delocalization logic, neither I did for the way apps are compiled. Therefore no specific tests to confirm access permissions were done.

Reply:
@Gvaihir Would you investigate whether such access is needed?

Reply:
@commandlinegirl , @emiloslavsky - could you explain how this test would look like? If you have no "Contribute" rights to the project, how can you compile a WF/make an applet there?

Reply:
It's about what project access the workflow's applet require at runtime, please check: https://documentation.dnanexus.com/developer/apps/app-metadata?q=dxapp.json#access-field-app-permissions

Reply:
Thanks! But that was my point - I did not change the way execs are compiled. Anyway, tested with extras.json (per @emiloslavsky suggestions). Added to integration tests, [16f8565](https://github.com/dnanexus/dxCompiler/pull/396/commits/16f8565e2b58765982bbbf9d8b1781a188f7abf5). No "CONTRIBUTE" access needed.